CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+# Workspace Loader
+
+## Devfile for workspace loader development
+
+[devfile.yaml](https://github.com/eclipse/che-workspace-loader/blob/master/devfile.yaml)
+
+## Developer workflow
+
+### Start a workspace from the devfile
+
+There are at least three options how to create and start a workspace from the devfile:
+
+- using [Eclipse Che CLI](https://github.com/che-incubator/chectl) on your own Eclipse Che installation:
+
+    ```bash
+    $ chectl workspace:start --devfile=https://raw.githubusercontent.com/eclipse/che-workspace-loader/master/devfile.yaml
+    ```
+
+- using factory loading way:
+
+    `https://<CheInstance>/f?url=https://raw.githubusercontent.com/eclipse/che-workspace-loader/master/devfile.yaml`
+
+- in User Dashboard from `Custom Workspace` page.
+
+As the result, Che Theia IDE will be opened and project will be cloned.
+
+### Install dependencies
+
+It's as simple as open `My Workspace` panel in Che Theia IDE and click on `[WS] install dependencies` task to run it. Or, just open terminal in `ws-loader-dev` container, navigate to the project directory and execute `yarn`:
+
+```bash
+# [ws-loader-dev container]
+$ cd /projects/che-workspace-loader && yarn
+```
+
+Once all the dependencies are installed you can make changes or proceed to the next step.
+
+### Build
+
+You may either run task `[WS] run build` from `My Workspace` panel or execute `yarn build` command in project directory:
+
+```bash
+# [ws-loader-dev container]
+$ cd /projects/che-workspace-loader && yarn build
+```
+
+### Run tests
+
+In order to run unit tests you need to find and run task `[WS] run tests` in `My workspace` panel or execute `yarn test` command in project directory:
+
+```bash
+# [ws-loader-dev container]
+$ yarn test
+```
+
+### Start development server
+
+The easiest way to start dev server is to run task `[WS] start dev server` from `My workspace` panel. Under the hood it executes following command in project directory:
+
+```bash
+# [ws-loader-dev container]
+$ yarn start --disable-host-check --public=$(echo ${server.dev-server} | sed -e s/https:\\/\\/// -e s/http:\\/\\/// -e s/\\///) --host="0.0.0.0" --env.target=${CHE_API_EXTERNAL%????}
+```
+
+### Testing new workspace loader
+
+When workspace loader server is run, you probably need to test introduced changes.
+
+To safely test the changes it is better to create a separate workspace which will be used by new loader.
+But by default, Che doesn't allow to run more than one workspace simultaneously.
+To change this behaviour you need to set `che.limits.user.workspaces.run.count` Che property to value greater than `1`.
+In development environment that could be reached by adding `CHE_LIMITS_USER_WORKSPACES_RUN_COUNT` environment variable for Che server deployment.
```

Comment:
```suggestion
In development environment that could be reached by adding `CHE_LIMITS_USER_WORKSPACES_RUN_COUNT` environment variable for Che server config map.
```
But since operator will rewrite these changes it makes sense to add just a reference to documentation where it's described how to change property depending on deployment method.

Also for developers, it makes sense to paste here simple commands:
```
kubectl patch configmap che --patch "{\"spec\":{\"data\":{\"CHE_LIMITS_USER_WORKSPACES_RUN_COUNT\": \"-1\"}}}" --type=merge -n che
kubectl rollout restart deployment/che -n che
```